src/Conn.cc

```diff
@@ -27,7 +27,7 @@ namespace zeek {
 uint64_t Connection::total_connections = 0;
 uint64_t Connection::current_connections = 0;
 
-Connection::Connection(const detail::ConnKey& k, double t, const ConnTuple* id, uint32_t flow, const Packet* pkt)
+Connection::Connection(const detail::ConnKeyPtr k, double t, const ConnTuple* id, uint32_t flow, const Packet* pkt)
```

Comment:
Taking the `ConnKeyPtr` as a non-reference here causes two ref-counts, one for the copy into the argument variable and one for the copy into the `key`. Arguably, this could either be a reference or you could use std::move for the assignment to `key`.